Upgrade virtual guest gives HTTP 500 error

When trying to update the network speed attribute of virtual guest from 10 to 100 it gives error
 Couldn't upgrade virtual guest: SoftLayer_Exception_Order_Item_Invalid: Invalid price 100 Mbps Public & Private Network Uplinks (Dedicated Host) (203859) provided on the order container. Please select a dedicated instance Uplink Port Speeds price to order a dedicated instance. (HTTP 500)

I was running the UAT "TestAccSoftLayerVirtualGuest_Basic" from https://github.com/softlayer/terraform-provider-softlayer
```TF_ACC=1 go test ./softlayer -v -run=TestAccSoftLayerVirtualGuest_Basic -timeout 1440m
=== RUN   TestAccSoftLayerVirtualGuest_Basic
The ID is 39351883
--- FAIL: TestAccSoftLayerVirtualGuest_Basic (236.63s)
	testing.go:280: Step 1 error: Error applying: 1 error(s) occurred:
		
		* softlayer_virtual_guest.terraform-acceptance-test-1: 1 error(s) occurred:
		
		* softlayer_virtual_guest.terraform-acceptance-test-1: Couldn't upgrade virtual guest: SoftLayer_Exception_Order_Item_Invalid: Invalid price 100 Mbps Public & Private Network Uplinks (Dedicated Host) (203859) provided on the order container. Please select a dedicated instance Uplink Port Speeds price to order a dedicated instance. (HTTP 500)
FAIL
exit status 1
FAIL	github.com/softlayer/terraform-provider-softlayer/softlayer	238.603s```

--- a/helpers/product/product.go
+++ b/helpers/product/product.go
@@ -19,12 +19,13 @@ package product
 import (
 	"fmt"
 
+	"strings"
+
 	"github.com/softlayer/softlayer-go/datatypes"
 	"github.com/softlayer/softlayer-go/filter"
 	"github.com/softlayer/softlayer-go/services"
 	"github.com/softlayer/softlayer-go/session"
 	"github.com/softlayer/softlayer-go/sl"
-	"strings"
 )
 
 // CPUCategoryCode Category code for cpus
@@ -128,7 +129,6 @@ func GetPackageProducts(
 // {"guest_core": 8.0, "ram": 32.0}
 // public[0] checks type of network.
 // public[1] checks type of cores.
-
 func SelectProductPricesByCategory(
 	productItems []datatypes.Product_Item,
 	options map[string]float64,
@@ -146,12 +146,18 @@ func SelectProductPricesByCategory(
 		forPublicCores = public[1]
 	}
 
+	forDedicatedHost := false
+	if len(public) > 2 {
+		forDedicatedHost = public[2]
+	}
+
 	// Filter product items based on sets of category codes and capacity numbers
 	prices := []datatypes.Product_Item_Price{}
 	priceCheck := map[string]bool{}
 	for _, productItem := range productItems {
 		isPrivate := strings.Contains(sl.Get(productItem.KeyName, "").(string), "PRIVATE")
 		isPublic := strings.Contains(sl.Get(productItem.Description, "Public").(string), "Public")
+		isDedicated := strings.Contains(sl.Get(productItem.KeyName, "").(string), "DEDICATED")
 		for _, category := range productItem.Prices[0].Categories {
 			for categoryCode, capacity := range options {
 				if _, ok := priceCheck[categoryCode]; ok {
@@ -173,11 +179,11 @@ func SelectProductPricesByCategory(
 				// Logic taken from softlayer-python @ http://bit.ly/2bN9Gbu
 				switch categoryCode {
 				case CPUCategoryCode:
-					if forPublicCores == isPrivate {
+					if forPublicCores == isPrivate || forDedicatedHost != isDedicated {
 						continue
 					}
 				case NICSpeedCategoryCode:
-					if forPublicNetwork != isPublic {
+					if forPublicNetwork != isPublic || forDedicatedHost != isDedicated {
 						continue
 					}
 				}

